File: uima-toolkit/src/main/scala/jenshaase/uimaScala/toolkit/type/Token_Type.scala
/* First created by JCasGenTue Apr 12 17:01:07 CEST 2011*/
package jenshaase.uimaScala.toolkit.type

import org.apache.uima.jcas.{JCas, JCasRegistry}
import org.apache.uima.cas.impl.{CASImpl, FSGenerator, TypeImpl}
import org.apache.uima.cas.{FeatureStructure, Type}

import org.apache.uima.jcas.tcas.Annotation_Type


/* Updated by JCasGen Tue Apr 12 17:01:07 CEST 2011
 * @generated */
class Token_Type extends org.apache.uima.jcas.tcas.Annotation_Type {

  /* @generated */
  def getFSGenerator() = {
    fsGenerator
  }

  /* @generated */
  val fsGenerator = 
    new FSGenerator() {
      def createFS(addr: Integer, cas: CASImpl): FeatureStructure = {
        if (Token_Type.this.useExistingInstance) {
          var fs = Token_Type.this.jcas.getJfsFromCaddr(addr)
        
          if (null == fs) {
            fs = new Token(addr, Token_Type.this)
            Token_Type.this.jcas.putJfsFromCaddr(addr, fs)
          }
          fs
        } else new Token(addr, Token_Type.this)
      }

    }

  /* initialize variables to correspond with Cas Type and Features
   * @generated */
  def this(jcas: JCas, casType: Type) = {
    this(jcas, casType)
    casImpl.getFSClassRegistry().addGeneratorForType(this.casType.asInstanceOf[TypeImpl], getFSGenerator())
  }


}

/* A simple token annotation
 * Updated by JCasGen Tue Apr 12 17:01:07 CEST 2011
 * @generated */
object Token_Type  {

  /* @generated */
  val typeIndexID = Token.typeIndexID

  /* @generated
   * @modifiable */
  val featOkTst = JCasRegistry.getFeatOkTst("jenshaase.uimaScala.toolkit.type.Token")


}

File: uima-toolkit/src/main/java/com/github/jenshaase/uimascala/toolkit/types/Token_Type.java

/* First created by JCasGen Sun Jan 22 13:52:21 CET 2012 */
package com.github.jenshaase.uimascala.toolkit.types;

import org.apache.uima.jcas.JCas;
import org.apache.uima.jcas.JCasRegistry;
import org.apache.uima.cas.impl.CASImpl;
import org.apache.uima.cas.impl.FSGenerator;
import org.apache.uima.cas.FeatureStructure;
import org.apache.uima.cas.impl.TypeImpl;
import org.apache.uima.cas.Type;
import org.apache.uima.cas.impl.FeatureImpl;
import org.apache.uima.cas.Feature;
import org.apache.uima.jcas.tcas.Annotation_Type;

/** 
 * Updated by JCasGen Sun Jan 22 13:52:21 CET 2012
 * @generated */
public class Token_Type extends Annotation_Type {
  /** @generated */
  protected FSGenerator getFSGenerator() {return fsGenerator;}
  /** @generated */
  private final FSGenerator fsGenerator = 
    new FSGenerator() {
      public FeatureStructure createFS(int addr, CASImpl cas) {
  			 if (Token_Type.this.useExistingInstance) {
  			   // Return eq fs instance if already created
  		     FeatureStructure fs = Token_Type.this.jcas.getJfsFromCaddr(addr);
  		     if (null == fs) {
  		       fs = new Token(addr, Token_Type.this);
  			   Token_Type.this.jcas.putJfsFromCaddr(addr, fs);
  			   return fs;
  		     }
  		     return fs;
        } else return new Token(addr, Token_Type.this);
  	  }
    };
  /** @generated */
  public final static int typeIndexID = Token.typeIndexID;
  /** @generated 
     @modifiable */
  public final static boolean featOkTst = JCasRegistry.getFeatOkTst("com.github.jenshaase.uimascala.toolkit.types.Token");
 
  /** @generated */
  final Feature casFeat_POS;
  /** @generated */
  final int     casFeatCode_POS;
  /** @generated */ 
  public String getPOS(int addr) {
        if (featOkTst && casFeat_POS == null)
      jcas.throwFeatMissing("POS", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return ll_cas.ll_getStringValue(addr, casFeatCode_POS);
  }
  /** @generated */    
  public void setPOS(int addr, String v) {
        if (featOkTst && casFeat_POS == null)
      jcas.throwFeatMissing("POS", "com.github.jenshaase.uimascala.toolkit.types.Token");
    ll_cas.ll_setStringValue(addr, casFeatCode_POS, v);}
    
  
 
  /** @generated */
  final Feature casFeat_lemma;
  /** @generated */
  final int     casFeatCode_lemma;
  /** @generated */ 
  public String getLemma(int addr) {
        if (featOkTst && casFeat_lemma == null)
      jcas.throwFeatMissing("lemma", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return ll_cas.ll_getStringValue(addr, casFeatCode_lemma);
  }
  /** @generated */    
  public void setLemma(int addr, String v) {
        if (featOkTst && casFeat_lemma == null)
      jcas.throwFeatMissing("lemma", "com.github.jenshaase.uimascala.toolkit.types.Token");
    ll_cas.ll_setStringValue(addr, casFeatCode_lemma, v);}
    
  
 
  /** @generated */
  final Feature casFeat_stem;
  /** @generated */
  final int     casFeatCode_stem;
  /** @generated */ 
  public String getStem(int addr) {
        if (featOkTst && casFeat_stem == null)
      jcas.throwFeatMissing("stem", "com.github.jenshaase.uimascala.toolkit.types.Token");
    return ll_cas.ll_getStringValue(addr, casFeatCode_stem);
  }
  /** @generated */    
  public void setStem(int addr, String v) {
        if (featOkTst && casFeat_stem == null)
      jcas.throwFeatMissing("stem", "com.github.jenshaase.uimascala.toolkit.types.Token");
    ll_cas.ll_setStringValue(addr, casFeatCode_stem, v);}
    
  



  /** initialize variables to correspond with Cas Type and Features
	* @generated */
  public Token_Type(JCas jcas, Type casType) {
    super(jcas, casType);
    casImpl.getFSClassRegistry().addGeneratorForType((TypeImpl)this.casType, getFSGenerator());

 
    casFeat_POS = jcas.getRequiredFeatureDE(casType, "POS", "uima.cas.String", featOkTst);
    casFeatCode_POS  = (null == casFeat_POS) ? JCas.INVALID_FEATURE_CODE : ((FeatureImpl)casFeat_POS).getCode();

 
    casFeat_lemma = jcas.getRequiredFeatureDE(casType, "lemma", "uima.cas.String", featOkTst);
    casFeatCode_lemma  = (null == casFeat_lemma) ? JCas.INVALID_FEATURE_CODE : ((FeatureImpl)casFeat_lemma).getCode();

 
    casFeat_stem = jcas.getRequiredFeatureDE(casType, "stem", "uima.cas.String", featOkTst);
    casFeatCode_stem  = (null == casFeat_stem) ? JCas.INVALID_FEATURE_CODE : ((FeatureImpl)casFeat_stem).getCode();

  }
}



    